### DISCUSSION TITLE
Added: C# Language Design Notes for Apr 11, 2017

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#650] [General]
[C# Language Design Notes for Apr 11, 2017](https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-04-11.md)

1. Runtime behavior of ambiguous default implementation

Please discuss.

**Comment by HaloFour** [comment]
> We should look more deeply into what Java does here. There must be accumulated insight already on this topic.

Java fails at runtime.

My concern with this is that Microsoft seems generally much more squeamish about the potential of breaking changes.  If adding a default interface member has the potential of causing an error at runtime due to a collision with some other interface I feel that the BCL team might feel that it's too risky to consider adding said members.  That effectively defeats the purpose of the proposal.
